docs/developer-docs/latest/developer-resources/error-handling.md

```diff
@@ -0,0 +1,95 @@
+- As a developer customizing the backend of your Strapi application, you could use controllers and services to [throw errors](#throwing-errors).
+
+## Receiving errors
+
+Errors are included in the response object with the `error` key and include information such as the HTTP status code, the name of the error, and additional information.
+
+### REST errors
+
+Errors thrown by the REST API are included in the [response](/developer-docs/latest/developer-resources/database-apis-reference/rest-api.html#unified-response-format) that has the following format:
+
+```json
+{
+  "data": null,
+  "error": {
+    "status": "", // HTTP status
+    "name": "", // Strapi error name ('ApplicationError' or 'ValidationError')
+    "message": "", // A human reable error message
+    "details": {
+      // error info specific to the error type
+    }
+  }
+}
+```
+
+<!-- TODO: add "types" (aka error `name`) list once settled -->
+
+### GraphQL errors
+
+Errors thrown by the GraphQL API are included in the [response](/developer-docs/latest/developer-resources/database-apis-reference/graphql-api.html#unified-response-format) that has the following format:
+
+```json
+{ "errors": [
+    {
+      "message": "", // A human reable error message
+      "extensions": {
+        "error": {
+          "name": "", // Strapi error name ('ApplicationError' or 'ValidationError'),
+          "message": "", // A human reable error message (same one as above);
+          "details": {}, // Error info specific to the error type
+        },
+        "code": "" // GraphQL error code (ex: BAD_USER_INPUT)
+      }
+    }
+  ],
+  "data": {
+    "graphQLQueryName": null
+  }
+}
+```
+
+## Throwing errors
+
+The recommended way to throw errors when developing any custom logic with Strapi is to have the [controller](/developer-docs/latest/development/backend-customization/controllers.md) respond with the correct status and body.
+
+This can be done by calling an error function on the context (i.e. `ctx`). Available error functions are listed in the [http-errors documentation](https://github.com/jshttp/http-errors#list-of-all-constructors) but their name should be camel-cased to be used by Strapi (e.g. `badRequest`).
+
+Error functions accept 2 parameters that correspond to the `error.message` and `error.details` attributes [received](#receiving-errors) by a developer querying the API:
+
+- the first parameter of the function is the error `message`
+- and the second one is the object that will be set as `details` in the response received
+
+```js
+
+// path: ./src/api/[api-name]/controllers/my-controller.js
+
+module.exports = {
+  exampleAction: async (ctx, next) => {
+    try {
+      ctx.body = 'ok';
+    } catch (err) {
+      // …
+      return ctx.badRequest('name is missing', { foo: 'bar' }).
+    }
+  }
+}
```

Comment:
I propose this example instead, as you mention the "try catch" use-case in the note below.

```js
module.exports = {
  renameDog: async (ctx, next) => {
    const newName = ctx.request.body.name;
    if (!newName) {
      return ctx.badRequest('name is missing', { foo: 'bar' })
    }
    ctx.body = strapi.service('api::dog.dog').rename(newName);
  }
}
```